data-processing/src/dfm_data/dolma_taggers/url.py

```diff
@@ -0,0 +1,106 @@
+"""
+Url taggers
+
+This module expands upon the collection of default dolma URL taggers
+"""
+
+import os
+from pathlib import Path
+
+from dolma.taggers.url import BaseDomainTagger
+from dolma.core.registry import TaggerRegistry
+
+def create_domain_blocklist_tagger(blocklist_path: Path) -> type[BaseDomainTagger]:
+    """Dynamically create taggers based on custom domain blocklist files specified in environment variable
+    The enironment variable is DOLMA_URL_CUSTOM_BLOCKLIST_PATH and is a comma-separated list of files.
+    Each file will spawn one tagger with the name custom_domain_blocklist_{path.stem}_v1
+    so it is important that each list of domains have different stem (filename)."""
+
+    class_name = f"CustomDomainBlocklistTagger{blocklist_path.stem}"
+    tagger_name = f"custom_domain_blocklist_{blocklist_path.stem}_v1"
+
+    # Build the class dynamically
+    cls = type(
+        class_name,
+        (BaseDomainTagger, ),
+        {
+            "BLOCKLIST_PATHS": [str(blocklist_path)]
+        },
+    )
+    # Add the class decorator explcitly to add the tagger to the registry
+    cls = TaggerRegistry.add(tagger_name)(cls)
+    return cls
+
+
+tagger_paths = os.environ.get("DOLMA_URL_CUSTOM_BLOCKLIST_PATH")
+
+if tagger_paths is not None:
+    for path in tagger_paths.split(","):
+        create_domain_blocklist_tagger(Path(path))
+
+@TaggerRegistry.add("adult_tld_v1")
```

Comment:
Generally have no issue running this filter to gain a list of domains, which we can the take the largest offenders from and then remove them or whitelist domains which contains these words but are not bad (I can do the validation if we have a public list). 

This can also help us establish a scale of the problem. Not sure if it is huge or not. I am even not sure it is something that we want to filter? I mean if someone want to write erotic literature then who am I to stop it.

My guess is however that there are smarter/more naive ways to do this (e.g. using removing sites which few people link to), which also filter poor quality.